mkosi/backend.py

```diff
@@ -571,19 +571,29 @@ class MkosiArgs:
 
     # Some extra stuff that's stored in MkosiArgs for convenience but isn't populated by arguments
     machine_id_is_fixed: bool
-    original_umask: int
     passphrase: Optional[Dict[str, str]]
 
+    output_split_root: Optional[Path] = None
+    output_split_verity: Optional[Path] = None
+    output_split_verity_sig: Optional[Path] = None
+    output_split_kernel: Optional[Path] = None
+
+    def architecture_is_native(self) -> bool:
+        return self.architecture == platform.machine()
+
+
+@dataclasses.dataclass
+class MkosiState:
+    """State related properties."""
+
+    original_umask: int
+
     output_checksum: Optional[Path] = None
     output_nspawn_settings: Optional[Path] = None
     output_sshkey: Optional[Path] = None
     output_root_hash_file: Optional[Path] = None
     output_root_hash_p7s_file: Optional[Path] = None
     output_bmap: Optional[Path] = None
```

Comment:
So these don't directly map to an option but they are derived from an option, so I would consider them config as well.